data-prepper-plugins/sns-sink/src/integrationTest/java/org/opensearch/dataprepper/plugins/sink/sns/SnsSinkServiceIT.java

```diff
@@ -108,45 +129,97 @@ private Collection<Record<Event>> setEventQueue(final int records) {
+    public SnsSinkService createObjectUnderTest(final String topicName) throws JsonProcessingException {
         String[] values = { topicName,region,stsRoleArn,dlqFilePath };
         final String configYaml = MessageFormat.format(SNS_SINK_CONFIG_YAML, values);
-        final SNSSinkConfig snsSinkConfig = objectMapper.readValue(configYaml, SNSSinkConfig.class);
-        return new SNSSinkService(snsSinkConfig,snsClient,pluginMetrics,pluginFactory,pluginSetting);
+        final SnsSinkConfig snsSinkConfig = objectMapper.readValue(configYaml, SnsSinkConfig.class);
+        return new SnsSinkService(snsSinkConfig,snsClient,pluginMetrics,pluginFactory,pluginSetting,mock(ExpressionEvaluator.class));
     }
 
     @ParameterizedTest
-    @ValueSource(ints = {1, 5, 10})
-    public void sns_sink_service_test_with_standard_queue_with_multiple_records(final int recordCount) throws JsonProcessingException {
-        final SNSSinkService objectUnderTest = createObjectUnderTest(standardTopic);
+    @ValueSource(ints = {5,9,10})
+    public void sns_sink_service_test_with_standard_queue_with_multiple_records(final int recordCount) throws JsonProcessingException, InterruptedException {
+        final SnsSinkService objectUnderTest = createObjectUnderTest(standardTopic);
         final Collection<Record<Event>> records = setEventQueue(recordCount);
+        final List<String> inputRecords = records.stream().map(Record::getData).map(Event::toJsonString).collect(Collectors.toList());
         objectUnderTest.output(records);
+        Thread.sleep(Duration.ofSeconds(10).toMillis());
+        List<String> topicData = readMessagesFromSNSTopicQueue(inputRecords,standardSqsQueue);
+        assertThat(inputRecords, is(topicData));
+        assertThat(inputRecords.size(), equalTo(topicData.size()));
         verify(snsSinkObjectsEventsSucceeded).increment(recordCount);
     }
 
+    @Test
+    public void sns_sink_service_test_with_standard_queue_with_multiple_batch() throws JsonProcessingException, InterruptedException {
+        final SnsSinkService objectUnderTest = createObjectUnderTest(standardTopic);
+            final Collection<Record<Event>> records = setEventQueue(11);
+        final List<String> inputRecords = records.stream().map(Record::getData).map(Event::toJsonString).collect(Collectors.toList());
+        objectUnderTest.output(records);
+        Thread.sleep(Duration.ofSeconds(10).toMillis());
+        List<String> topicData = readMessagesFromSNSTopicQueue(inputRecords,standardSqsQueue);
+        assertThat(inputRecords, is(topicData));
+        assertThat(inputRecords.size(), equalTo(topicData.size()));
+        verify(snsSinkObjectsEventsSucceeded,times(2)).increment(anyDouble());
+    }
+
+    private List<String> readMessagesFromSNSTopicQueue(List<String> inputRecords, final String sqsQueue) {
+        final List<Message> messages = new ArrayList<>();
+        do{
```

Comment:
This should use Awaitility to wait for a maximum period of time. Perhaps 1 minute.

As it is, this can get stuck indefinitely.